wallet/wallet.c

```diff
@@ -1414,7 +1411,7 @@ bool wallet_init_channels(struct wallet *w)
 
 static enum channel_state_db get_state_channel_db(const char *dir, const char *typ)
 {
-	for (int i = 0; i < CHANNEL_STATE_DB_SIZE; i++) {
+	for (size_t i = 0; i < ARRAY_SIZE(channel_state_params); i++) {
```

Comment:
it'd be better to fold this change in with the commit that originally made the line.